design-proposals/eim-cli.md

```diff
@@ -0,0 +1,254 @@
+
+## Workflows
+
+The following workflows are to be supported initially - the list may be expanded in the future if necessary functionality needs to be supported via CLI. It is assumed that the Bulk Import Tool [(BIT)](https://github.com/open-edge-platform/infra-core/tree/main/bulk-import-tools) will be integrated as part of the CLI effort.
+
+> **NOTE**: The commands and the order of commands at this stage is in no way final - they are just placeholders until we agree on actual command structure overall.
+
+### Ability to register/onboard/provision a host
+
+#### Local Options
+
+The local options associated with the `host` object/noun would help with filtering and providing arguments to the commands:
+
+- `-o` `registered`/`onboarded`/`provisioned`/`all`/`deauthorized`/`unknown`/`error` - helps filter out the output of `list` command/verb ( TODO maybe instead of `-o` for output change to `-s` for status???)
+- `--import-from-csv` `my-hosts.csv` - used to provision a bulk of hosts using BIT
+> **NOTE**: TODO - Team I am not yet convinced on using the register/onboard/provision flags as options to the create/delete objects/nouns rather than using as verbs directly - will welcome feedback
+- `--register` `<arguments>` - local flag for the `create` command - enables only registering host during create - takes in argument in particular order necessary for registering the host 
+- `--onboard` - local flag for the `create` command - enables only onboarding of already registered host during create
+- `--provision` `<arguments>` - local flag for the `create` command - enables only provisioning of already onboarded host during create - takes in argument in particular order necessary for provisioning of the host
+- `--auto-provision` - optional local flag that only takes effect when `--register` flag is invoked - enables auto provisioning of node when it connects
+> **Note**: TODO - If it is the case that the autoprovision toggle in registration view in UI is ignored if the provisioner is set to autoprovision (TODO also get to to bottom why the toggle is allowed in UI if that is the case) then the CLI should spit out a warning message about this whenever a host is being created/registered/onboarded
+- `--auto-onboarding` - optional local flag that only takes effect when `--register` flag is invoked - enables auto onboarding of node when it connects
+- `--deauthorize-only` - optional local flag that only takes effect when `delete` flag is invoked - does not delete host fully but only deauthorizes it
+- `--force-delete` - optional local flag that only takes effect when `delete` flag is invoked - deletes host without deauthorization
+- `--edit-config` `<arguments>` - set of arguments to aid the edition of host
```

Comment:
this could be a can of worms but we should be able to intercept when the object is immutable by looking at the returned error code